font

```diff
@@ -158,3 +158,40 @@ sub on_user_command
     $self->change_size(($1 eq "increment") ? +1 : -1);
   }
 }
+
+sub on_init
+{
+  my ($self) = @_;
+
+  ($self->{winx}, $self->{winy}) = (0, 0);
+}
+
+sub get_dpi_at
+{
+  my ($px, $py) = @_;
+
+  foreach (grep {m/ connected /} `xrandr`) {
+    my ($w, $h, $x, $y, $wmm, $hmm) = m/\b(\d+)x(\d+)([-+]\d+)([-+]\d+) .* (\d+)mm x (\d+)mm\b/;
+    next if $px < $x or $py < $y or $px >= $x + $w or $py >= $y + $h;
+    # Calculate DPI based on width alone for now
+    return $w * 25.4 / $wmm;
```

Comment:
I think it might be clearer to write:
```
mm_per_inch = .0393701
return $w/($wmm*$mm_per_inch)
```

Or similar.  And maybe leave a comment for future readers about how ppi/dpi is calculated.